lib/spack/spack/test/concretize.py

```diff
@@ -455,22 +459,72 @@ def test_concretize_two_virtuals_with_dual_provider_and_a_conflict(self):
         with pytest.raises(spack.error.SpackError):
             s.concretize()
 
+    @pytest.mark.only_clingo(
+        "Optional compiler propagation isn't deprecated for original concretizer"
+    )
+    @pytest.mark.parametrize("unify", [True, False, "when_possible"])
+    def test_concretize_environment_propagated_disabled_variant(
+        self, unify, tmpdir, mutable_mock_env_path
+    ):
```

Comment:
This test instead doesn't seem to fail on `develop`, so it is either wrong or not a reproducer.